src/components/TopBar/Wallet/OhmFaucet.tsx

```diff
@@ -0,0 +1,36 @@
+import { t } from "@lingui/macro";
+import { Box, useTheme } from "@material-ui/core";
+import { SecondaryButton } from "@olympusdao/component-library";
+import { useDispatch, useSelector } from "react-redux";
+import { FAUCET_PENDING_TYPE, getOhm, hasOhmFaucet } from "src/helpers/OhmFaucet";
+import { useWeb3Context } from "src/hooks";
+import { State } from "src/slices/interfaces";
+import { isPendingTxn, txnButtonText } from "src/slices/PendingTxnsSlice";
+
+export const OhmFaucetButton = () => {
+  const theme = useTheme();
+  const dispatch = useDispatch();
+  const { address, provider, networkId } = useWeb3Context();
+
+  const pendingTransactions = useSelector((state: State) => {
+    return state.pendingTransactions;
+  });
+
+  const runOhmFaucet = async () => {
+    dispatch(getOhm({ address, provider, networkID: networkId }));
```

Comment:
I don't want to hold you up, but if this isn't time sensitive can you use `react-query`?

Reply:
It's not time-sensitive - it was code that I had lying around for #657 , and I thought it would be useful for others. 

I will integrate `react-query` into this.